src/security.ts

```diff
@@ -167,3 +170,60 @@ export const iamMfaEnabledForConsoleAccess: ResourceValidationPolicy = {
     }),
 };
 registerPolicy("iamMfaEnabledForConsoleAccess", iamMfaEnabledForConsoleAccess);
+
+
+/** @internal */
+// Enforce the note on aws.iam.Role:
+//
+// NOTE: If you use this resource’s managed_policy_arns argument or inline_policy configuration blocks, this resource
+// will take over exclusive management of the role's respective policy types (e.g., both policy types if both arguments
+// are used). These arguments are incompatible with other ways of managing a role's policies, such as
+// aws.iam.PolicyAttachment, aws.iam.RolePolicyAttachment, and aws.iam.RolePolicy. If you attempt to manage a role’s
+// policies by multiple means, you will get resource cycling and/or errors.
+export const iamRoleNoPolicyManagementConflicts: StackValidationPolicy = {
+    name: "iam-role-no-policy-management-conflicts",
+    description: "Checks that iam.Role resources do not conflict with iam.PolicyAttachment, iam.RolePolicyAttachment, iam.RolePolicy",
+    validateStack: (args: StackValidationArgs, reportViolation: ReportViolation) => {
+        args.resources.forEach(r => {
+            var roleProp: string;
+            var currentType: string;
+            switch(r.type) {
+                case "aws:iam/policyAttachment:PolicyAttachment": {
+                    roleProp = "roles";
+                    currentType = "PolicyAttachment";
+                    break
+                }
+                case "aws:iam/rolePolicyAttachment:RolePolicyAttachment": {
+                    roleProp = "role";
+                    currentType = "RolePolicyAttachment";
+                    break
+                }
+                case "aws:iam/rolePolicy:RolePolicy": {
+                    roleProp = "role";
+                    currentType = "RolePolicy";
+                    break
+                }
+                default: {
+                    return
+                }
+            }
+
+            if (r.propertyDependencies[roleProp]) {
```

Comment:
I suppose this won't work if a role is defined in one stack and policy attachments in another? Or is there some way to traverse stack dependencies?